inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/project/KnowledgeBaseDetailsPanel.java

```diff
@@ -594,5 +606,29 @@ private void addUrlField(WebMarkupContainer wmc, String id)
             textField.add(Validators.URL_VALIDATOR);
             wmc.add(textField);
         }
+
+        private CheckBox maxQueryLimitCheckbox(String id, IModel<Boolean> model) {
+            return new AjaxCheckBox(id, model) {
```

Comment:
The Checkbox is just used to set the value of the ``queryLimitFied`` to Integer.MAX_VALUE. It's value is not safed.

Reply:
You mean before setting a limit, we have to check the checkbox? IMHO having a limit field with a default value (whatever is configured in the `KnowledgeBaseProperties` / `settings.properties` is sufficient. Having an extra checkbox to enable the limit only adds complexity to the UI. In any case, `Integer.MAX_VALUE` would definitely be too much. Either the KB is smallish and doesn't hit the limit anyway or the KB is large and then the limit needs to be chosen carefully (not too large).

Reply:
When the checkbox is clicked, the value of the queryLimitField is set to Integer.MAX_VALUE.
It is supposed to basically "remove" the limit.
> I'd like to be able to specify that I want all results to be retrieved

Reply:
If you try that on a large KB, it may crash the application or simply time out. On a small KB, you could just manually set the limit to something very large. Btw. theoretically a KB could return more than max-int results - so max-int doesn't necessarily mean "all".

Reply:
> Btw. theoretically a KB could return more than max-int results - so max-int doesn't necessarily mean "all".

I know that. I thought that *realistically*, max-int means "all" for all knowledge bases that will be used in Inception.

Even without the checkbox the user can set the value to something that is too large and may cause problems. 
The user can also set it to a smaller value if it causes problems with max-int.
If we want to further restrict the possible values for the Limit what would be a reasonable MAX value?

Reply:
That probably depends on the resources that the server admin gives to INCEpTION and on the usage pattern that the admin expects (e.g. number of concurrent users, number/size of KBs being used, etc.). So in principle a second setting like `hardMaxResults` in the `settings.properties` where the admin can configure a hard upper limit would probably be reasonable.

That said, these kinds of limits are not a panacea. In some cases, they are probably ok (e.g. in the auto-complete dropdown). In other cases, it would be preferable not to have a limit at all, but instead tackle the problem using paging (e.g. when listing instances on the KB page).